staging/src/k8s.io/client-go/util/jsonpath/parser.go

```diff
@@ -334,13 +334,23 @@ Loop:
 func (p *Parser) parseFilter(cur *ListNode) error {
 	p.pos += len("[?(")
 	p.consumeText()
+	quotes := 0
 Loop:
 	for {
 		switch p.next() {
 		case eof, '\n':
 			return fmt.Errorf("unterminated filter")
+		case '"':
```

Comment:
hi @mtaufen after I check the code [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/util/printing.go#L111) : 
I think currently `GetFlagString(cmd, "output")` this function only support single quotes, if you using double quotes the `outputFormat`  will be like this 
```console 
(string) (len=78) "jsonpath={.items[?(@.status.nodeInfo.osImage == Deb)].status.nodeInfo.osImage}"
```
instead of normal one
```console
(string) (len=80) "jsonpath={.items[?(@.status.nodeInfo.osImage == \"Deb\")].status.nodeInfo.osImage}"
```
I would like to file another issue to track if you think `GetFlagString` function should change to a more suitable, and take a deeper look, wdyt ?